### ISSUE TITLE
pop3 tests don't cover both deprecated and undeprecated IMailbox behavior codepaths

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#2003]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#2003|
|Type|defect|
|Created|2006-08-18 02:58:02Z|

listMessages can raise a ValueError or an IndexError, but IndexError is about to be deprecated.  The tests should cover both of these cases and suppress warnings from the deprecated one.

<details><summary>Searchable metadata</summary>

```
trac-id__2003 2003
type__defect defect
reporter__exarkun exarkun
priority__highest highest
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__mail mail
keywords__ 
time__1155869882000000 1155869882000000
changetime__1155999392000000 1155999392000000
version__None None
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_1"></a>|@exarkun removed owner|
|-|-|

Read for review in pop3-indexerror-tests-2003

**Comment by twisted-trac** [comment]
|[<img alt="jerub's avatar" src="https://avatars.githubusercontent.com/u/315461?s=50" width="50" height="50">](https://github.com/jerub)<a name="note_2"></a>|@jerub set owner to @exarkun|
|-|-|

The level of effort you've gone to with the tests is amazing. Thankyou for your attention to detail. I don't especially like how you've copied the exact, long, string for the deprecation warning, but with deprecation warnings, it's for the best.

Please merge.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_3"></a>|@exarkun set status to `closed`|
|-|-|

(In [17856]) Merge pop3-indexerror-tests-2003

Author: exarkun
Reviewer: jerub
Fixes [#2003](https://github.com/twisted/twisted/issues/2003)

This adds test coverage for the non-deprecated error handling codepaths in the
POP3 server.  It also adds suppressions for the pending deprecation warnings
emitted from old codepaths.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_4"></a>|@exarkun commented|
|-|-|

(In [17859]) Merge pop3-indexerror-tests-2003

Author: exarkun
Reviewer: jerub
Fixes [#2003](https://github.com/twisted/twisted/issues/2003)

This adds test coverage for the non-deprecated error handling codepaths in the
POP3 server.  It also adds suppressions for the pending deprecation warnings
emitted from old codepaths.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
